### PR TITLE
DM-47606: Initial code for s3proxy.

### PR DESCRIPTION
**Pull request by ktlim**

### CODE REVIEW

File: src/s3proxy/handlers/external.py [hunk 1]
```diff
@@ -50,3 +53,33 @@ async def get_index(
...
+    This returns the contents of an s3 object
+    """
+    path = f"s3://{bucket}/{key}"
+    logger.info("s3 request")
+    rp = ResourcePath(path)
+    mimetype, encoding = mimetypes.guess_type(path)
+    if mimetype is None:
+        mimetype = "application/octet-stream"
+    try:
+        return Response(content=rp.read(), media_type=mimetype)
```
Comment (athornton) [inline src/s3proxy/handlers/external.py:81]: Are the objects always going to be small enough that we don't need to care about streaming them?

File: tox.ini [hunk 2]
```diff
@@ -34,3 +34,7 @@ commands = pre-commit run --all-files
 description = Run the development server with auto-reload for code changes.
 usedevelop = true
 commands = uvicorn s3proxy.main:app --reload
+pass_env =
+    LSST_RESOURCES_S3_PROFILE_*
```
Comment (athornton) [inline tox.ini:38]: Where does this get used, and what does it do?

**Review by athornton — APPROVED** [review]
Looks OK, as long as you've considered whether the response contents will always fit into memory and therefore we don't need to stream them.

I also don't understand whether `LSST_RESOURCES_S3_PROFILE_*` is for future profile customization, because it doesn't look to me like it's in use.

File: tox.ini (hunk 2 above)
Comment (timj) [reply tox.ini:38]: `lsst.resources` uses that to work out which S3 profile is meant to be used for auth.

File: src/s3proxy/handlers/external.py (hunk 1 above)
Comment (ktlim) [reply src/s3proxy/handlers/external.py:81]: DM-47720 filed for this.  For now, everything should fit just fine.

**Pull request merged**